source/routing/raptor_solution_reader.cpp

```diff
@@ -88,7 +88,14 @@ get_out_st_dt(const std::pair<const type::StopTime*, DateTime>& in_st_dt,
     return {nullptr, 0};
 }
 
-bool is_valid(const Journey& j) {
+template<typename Visitor>
+bool is_valid(const RaptorSolutionReader<Visitor>& reader, const Journey& j) {
```

Comment:
In this function, the only thing you need is `reader`'s `departure_datetime`. For me, it would be more readable (and avoid the template) if we pass only `reader.departure_datetime` than pass the entire `reader`. On the other hand, if we want to unit-test this function, we won't have to construct a complete `reader` for testing purpose.